### ISSUE TITLE
Deployment Report: Variable referenced before assignment

### ISSUE THREAD

**Issue by benformosa** [sassoftware/viya-ark#6]
Variables `java_memory` and `h_read_memory` may be referenced before assignment.

When running the Get SAS host details task for a CAS controller, the module fails with one of the following errors:

[java_memory](https://github.com/sassoftware/viya-ark/blob/master/playbooks/deployment-report/library/get_sas_host_details.py#L1249)
```
fatal: [cascontroller0]: FAILED! => {"changed": false, "module_stderr": "Shared connection to hostname.example.com closed.", "module_stdout": "Traceback (most recent call last):
  File \"/tmp/.satellite.ansible/ansible-tmp-1556662690.57-280062909158338/AnsiballZ_get_sas_host_details.py\", line 113, in <module>
    _ansiballz_main()
  File \"/tmp/.satellite.ansible/ansible-tmp-1556662690.57-280062909158338/AnsiballZ_get_sas_host_details.py\", line 105, in _ansiballz_main
    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)
  File \"/tmp/.satellite.ansible/ansible-tmp-1556662690.57-280062909158338/AnsiballZ_get_sas_host_details.py\", line 48, in invoke_module
    imp.load_module('__main__', mod, module, MOD_DESC)
  File \"/tmp/ansible_get_sas_host_details_payload_mtxjsu/__main__.py\", line 1708, in <module>
  File \"/tmp/ansible_get_sas_host_details_payload_mtxjsu/__main__.py\", line 964, in main
  File \"/tmp/ansible_get_sas_host_details_payload_mtxjsu/__main__.py\", line 1249, in _get_sas_service_info
UnboundLocalError: local variable 'java_memory' referenced before assignment", "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error", "rc": 1}
```

[h_read_memory](https://github.com/sassoftware/viya-ark/blob/master/playbooks/deployment-report/library/get_sas_host_details.py#L1263)
```
fatal: [cascontroller0]: FAILED! => {"changed": false, "module_stderr": "Shared connection to hostname.example.com closed.", "module_stdout": "Traceback (most recent call last):
  File \"/tmp/.satellite.ansible/ansible-tmp-1556663118.37-225135293679002/AnsiballZ_get_sas_host_details.py\", line 113, in <module>
    _ansiballz_main()
  File \"/tmp/.satellite.ansible/ansible-tmp-1556663118.37-225135293679002/AnsiballZ_get_sas_host_details.py\", line 105, in _ansiballz_main
    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)
  File \"/tmp/.satellite.ansible/ansible-tmp-1556663118.37-225135293679002/AnsiballZ_get_sas_host_details.py\", line 48, in invoke_module
    imp.load_module('__main__', mod, module, MOD_DESC)
  File \"/tmp/ansible_get_sas_host_details_payload_9HkCy0/__main__.py\", line 1709, in <module>
  File \"/tmp/ansible_get_sas_host_details_payload_9HkCy0/__main__.py\", line 964, in main
  File \"/tmp/ansible_get_sas_host_details_payload_9HkCy0/__main__.py\", line 1264, in _get_sas_service_info
UnboundLocalError: local variable 'h_read_memory' referenced before assignment", "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error", "rc": 1}
```

**Comment by erharb** [comment]
@sasjowood I can't recall did you run into this issue and fix it recently in another merge internally?

**Comment by erharb** [comment]
Nevermind this was unrelated, we'll try to get this fix in soon

**Comment by kevinlinglesas** [comment]
@benformosa thanks for pointing this out.  We had actually determined the problem existed and were essentially ready with a fix.  This issue should be resolved by [Viya34-akrk-1.2](https://github.com/sassoftware/viya-ark/releases/tag/Viya34-ark-1.2 , ), Release note: DEPENB-1713.
